src/components/APIInterface.js

```diff
@@ -0,0 +1,88 @@
+import { fetchAndParse } from "../fetchAndParse";
+
+export class APIInterface {
```

Comment:
Also, if this is an Official Object-Oriented Programming Interface™, then we should have an interface that just defines the empty methods and has all the documentation for what they take and what they return, and a separate class that *implements* the interface by inheriting from the first class and having the code to make the web requests, like `export class ServerAPI extends APIInterface`. All the React component code would only talk about the interface, and never mention the specific implementation.

Then I can write *another* implementation of the interface that uses local computation (sometimes) instead of remote web requests, and it will be drop-in compatible because both implementations implement the same interface. [Barbara Liskov will smile upon us](https://en.wikipedia.org/wiki/Liskov_substitution_principle) and everything will Just Work without me having to make any more changes to the React components.